File: LeetCode/python/SingleNumber.py
class Solution:
    # @param {integer[]} nums
    # @return {integer}
    def singleNumber(self, nums):
        #利用异或来求：如果两个数相同，那么会变成0；那么题目中的所有相同的数对就会变成0，最终剩下的就是那个单数
        result = 0
        for num in nums:
            result ^= num
		
        return result
		
		
#测试程序
a = [1,1,2,2,3,8,8]
s = Solution()
print(s.singleNumber(a))

File: LeetCode/iter2/C++/SingleNumber.cpp
#include <iostream>
#include <string>
#include <algorithm>
#include <vector>
#include <sstream>
#include <unordered_map>
#include <unordered_set>
#include <utility>
#include <queue>
#include <stack>

using namespace std;

class Solution {
public:
	int singleNumber(vector<int>& nums) {
		int ret=0;
		for (int num : nums)
			ret ^= num;
		
		return ret;
	}
};


int main(void)
{
	vector<int> v;
	v.push_back(0);
	v.push_back(1);
	v.push_back(3);
	v.push_back(0);
	v.push_back(1);
	v.push_back(3);
	v.push_back(2);

	Solution s;
	cout << s.singleNumber(v)<<endl;

	return 0;
}